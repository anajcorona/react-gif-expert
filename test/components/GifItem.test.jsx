const { render, screen } = require("@testing-library/react")
const { GifItem } = require("../../src/components/GifItem")

describe('Pruebas en <GifItem/>', () => { 
    const title = 'Dragonball';
    const url = 'https://one-punch.com/'
    test('Debe hacer march con el snapshowt ', () => { 
        const{ container } = render( <GifItem title={title} url={url}/> );
        expect(container).toMatchSnapshot();
     })

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render( <GifItem title={ title} url={url} />);
        //expect ( screen.getByRole('img').src ).toBe(url);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    })

    test('Debe mostrar el titulo en el componente', () => { 
        render( <GifItem title={title} url={ url}/>);
        expect( screen.getByText( title )).toBeTruthy();
    })
 })