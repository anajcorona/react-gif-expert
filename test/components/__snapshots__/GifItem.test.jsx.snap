// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe hacer march con el snapshowt  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragonball"
      src="https://one-punch.com/"
    />
    <p>
       
      Dragonball
       
    </p>
  </div>
</div>
`;
